### DISCUSSION TITLE
Any way to change "Idle timeout" of ALB in Load Balanced Web Service?

### DISCUSSION THREAD

**Discussion by toshitanian** [aws/copilot-cli#3632] [Q&A] [answered]
Hi, 

I'm working on deploying an application using copilot "Load Balanced Web Service". In that application, one of the endpoint takes several minutes to response. In this case, the ALB responses `504 Gateway TImeout` doe to the timeout.
I found that the "idle timeout" is set to 60 seconds in the ALB the copilot creates. According to [the document](https://docs.aws.amazon.com/elasticloadbalancing/latest/application/application-load-balancers.html#connection-idle-timeout), we can change the idle timeout from 1 second to 4000 seconds.

So, I looked into the document of copilot CLI, but was not able to find the section for the idle timeout. Is there any regular way to set the idle timeout or some workaround? If I need a workaround, I'm wondering whether I can do patching with addon cfn.
https://aws.github.io/copilot-cli/ja/docs/manifest/lb-web-service/

Thank you in advance

**Comment by paragbhingre** [comment]
Hello @toshitanian, 

Thank you for reporting your concern with us. Today, Copilot does not support modifying the value of "idle timeout" parameter. Also, addon cfn will not allow it as it's purpose is to add new resources (e.g. s3 bucket, security groups etc.) to your existing service. But we have made a note of your requirement and we will consider it.

**Reply by paragbhingre** [reply]
But meanwhile you can use [modify-load-balancer-attributes](https://docs.aws.amazon.com/cli/latest/reference/elbv2/modify-load-balancer-attributes.html) AWS API to change the `idle_timeout.timeout_seconds` attribute. To get your `load-balancer-arn` you could run `copilot env show --resources` and look for the load balancer resource. Let me know if you need guidance with anything else.

**Reply by toshitanian** [reply]
@paragbhingre
Thank you for the reply! Will update manually using CLI or from the web console 👍

**Reply by mebibou** [reply]
Any of you could share a script to do this in the meantime? it would be very useful to be able to do it all from the yaml config but a script would be helpful as well

**Comment by brianmcburney** [comment, marked as answer]
Sharing this solution here for others who come across this discussion: https://github.com/aws/copilot-cli/issues/4719
